Sources/StreamChat/APIClient/CDNClient_Mock.swift

```diff
@@ -5,21 +5,23 @@
 import StreamChat
 import StreamChatTestTools
 
-final class CDNClient_Mock: CDNClient {
+final class CDNClient_Mock: CDNClient, Spy {
+    var recordedFunctions: [String] = []
+
     static var maxAttachmentSize: Int64 { .max }
-    
-    lazy var uploadAttachmentMockFunc = MockFunc.mock(for: uploadAttachment)
+    var uploadAttachmentProgress: Double?
+    var uploadAttachmentResult: Result<URL, Error>?
+
     func uploadAttachment(
         _ attachment: AnyChatMessageAttachment,
         progress: ((Double) -> Void)?,
         completion: @escaping (Result<URL, Error>) -> Void
     ) {
-        uploadAttachmentMockFunc.callAndReturn(
-            (
-                attachment,
-                progress,
-                completion
-            )
-        )
+        record()
+        uploadAttachmentProgress.map { progress?($0) }
+
+        DispatchQueue.main.asyncAfter(deadline: .now() + 0.1) {
+            self.uploadAttachmentResult.map(completion)
+        }
```

Comment:
nit: I'd prefer to save the `completion` and call it manually, tbh. It just makes debugging easier when the stack trace clearly shows what the sequence of calls is